optaplanner-benchmark/src/main/java/org/optaplanner/benchmark/impl/report/BenchmarkReport.java

```diff
@@ -356,7 +356,7 @@ private void writeBestScoreSummaryChart() {
                     double[] levelValues = ScoreUtils.extractLevelDoubles(singleBenchmarkResult.getAverageScore());
                     for (int i = 0; i < levelValues.length && i < CHARTED_SCORE_LEVEL_SIZE; i++) {
                         if (isFinite(levelValues[i])) {
-                            if (i >= datasetList.size()) {
+                            while (i >= datasetList.size()) {
```

Comment:
This doesn't entirely fix the bug:

Presume levelValues.length = 5
So i goes 0, 1, 2, 3, 4
Now if isFinite(levelValues[2]) returns false, but the others return true, then this will work properly.
However if isFinite(levelValues[4]) returns false (the last index), then this won't work properly.

Reply:
If the last value `isFinite(levelValues[4])` returns false, then the loop will exit without adding the value. I don't see an out of bounds access here. Maybe I'm not getting you, is there a consequence in this corner case I'm not seeing?